discretisedfield/field.py

```diff
@@ -3230,10 +3259,39 @@ def _fromovf(cls, filename):
         r_tuple = (*reversed(mesh.n), header["valuedim"])
         t_tuple = (2, 1, 0, 3)
 
+        try:
+            components = header["valuelabels"].split()
+        except KeyError:
+            components = None
+        else:
+            if "_" in components[0]:  # OOMMF writes: Magnetization_x
+                components = [c.split("_")[1] for c in components]
+            if len(components) != len(set(components)):  # components are not unique
+                components = None
+
+        try:
+            unit_list = header["valueunits"].split()
+        except KeyError:
+            units = None
+        else:
+            if len(unit_list) == 0:
+                units = None  # no unit in the file
+            elif len(set(unit_list)) != 1:
```

Comment:
We are not checking the same if a user directly assigns multiple units to a field like `df.Field(mesh, dim=3, value=(1, 1, 1), units="m s kg")`. Do you think we should check for multiple units at the time of initialization as well?

Reply:
For the OOMMF generated files we know that we can get the same unit multiple (typically three) times separated by spaces and we only want to keep it one time. For a general field we could expect a situation in which a user wants to define a "more complicated" unit consisting of multiple SI units and decide to separate theses with spaces (instead of asterisks) - which should be fine as the units are basically only used for plots generated via `xarray`. That's why I didn't add any more strict tests for the user input. [I expect (and hope) that OOMMF would never write a single combined unit with a space in it.]

Reply:
I see. Sounds reasonable.